token_utils/unpack.py

```diff
@@ -0,0 +1,46 @@
+"""
+Token unpacking and verifying functions.
+"""
+
+from time import time
+
+from django.conf import settings
+from jwkest import Expired, Invalid, jwk
+from jwkest.jws import JWS
+
+
+def unpack_jwt(token, lms_user_id, now=None):
+    """
+    Unpack and verify an encoded JWT.
+
+    Validate the user, expiration, and block.
+
+    Arguments:
+        token (string): The token to be unpacked and verified.
+        lms_user_id (int): LMS user ID this token should match with.
+        now (int): Optional now value for testing.
+    """
+    now = now or int(time())
+    payload = _unpack_and_verify(token)
+
+    # todo: are these the correct exceptions to throw?
+    # todo: should all relevant exceptions be thrown instead of just 1?
```

Comment:
Question: This is just throwing the first exception. Should it throw multiple instead if both are invalid?

Reply:
diagnosing all the problems at once is a niceness thing for the user - in general you throw a set of all problems together to help the user fix them. But it's fiddlier to build so it has to be useful, I'd say leave it out of the first version.

Reply:
That makes sense!